tests/utils/set_params.cpp

```diff
@@ -222,16 +222,31 @@ void setInvertParam(QudaInvertParam &inv_param)
   inv_param.use_alternative_reliable = alternative_reliable;
   inv_param.use_sloppy_partial_accumulator = 0;
   inv_param.solution_accumulator_pipeline = solution_accumulator_pipeline;
-  inv_param.max_res_increase = 1;
+  inv_param.max_res_increase = max_res_increase;
+  inv_param.max_res_increase_total = max_res_increase_total;
 
   // domain decomposition preconditioner parameters
   inv_param.inv_type_precondition = precon_type;
 
   inv_param.schwarz_type = precon_schwarz_type;
+
+  inv_param.madwf_diagonal_suppressor = madwf_diagonal_suppressor;
+  inv_param.madwf_ls = madwf_ls;
+
+  inv_param.madwf_null_miniter = madwf_null_miniter;
+  inv_param.madwf_null_tol = madwf_null_tol;
+  inv_param.madwf_train_maxiter = madwf_train_maxiter;
+
+  inv_param.madwf_param_load = madwf_param_load ? QUDA_BOOLEAN_TRUE : QUDA_BOOLEAN_FALSE;
+  inv_param.madwf_param_save = madwf_param_save ? QUDA_BOOLEAN_TRUE : QUDA_BOOLEAN_FALSE;
+
+  strcpy(inv_param.madwf_param_infile, madwf_param_infile);
+  strcpy(inv_param.madwf_param_outfile, madwf_param_outfile);
+
   inv_param.precondition_cycle = precon_schwarz_cycle;
   inv_param.tol_precondition = tol_precondition;
   inv_param.maxiter_precondition = maxiter_precondition;
-  inv_param.verbosity_precondition = mg_verbosity[0];
+  inv_param.verbosity_precondition = verbosity_precondition;
```

Comment:
Does this change affect the verbosity settings when running multigrid?

Reply:
This is in the `setInvertParam` so I think it is irrelevant for MG? And this is only for tests so should not affect the general behavior?

Reply:
I wholeheartedly agree with this change, it actually caused me some confusion while working on the local version of the Schur op. If I had noticed this when it was originally introduced I would've flagged it. This is dealt with appropriately for the MG codepath iirc, where a separate `inv_param` structure is created.